compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -58,19 +58,42 @@ static FailureOr<Operation *> getRootOp(func::FuncOp funcOp) {
   return rootOp;
 }
 
+/// Builds a proper tile sizes vector for the op.
+/// scf::tileUsingSCFForOp expects the num of tile sizes = num of loops. This
+/// method returns a proper tile sizes vector for each op during tiling.
+static SmallVector<Value> buildTileSizesForOp(OpBuilder &b, Operation *op,
+                                              SmallVector<int64_t> tileSizes) {
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(op);
```

Comment:
I think you can cast this to `TilingInterface` and assert. You need to only check that the op implements the tiling interface. Also I dont think you need to change the insertion point (but there is no harm in it).

Reply:
Done. Using the current insertion point but still restore it with the guard.